tests/rptest/services/franz_go_verifiable_services.py

```diff
@@ -0,0 +1,198 @@
+# Copyright 2022 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import os
+import threading
+from ducktape.services.background_thread import BackgroundThreadService
+
+# The franz-go root directory
+TESTS_DIR = os.path.join("/opt", "si-verifier")
+
+from enum import Enum
+
+
+class ServiceStatus(Enum):
+    SETUP = 1
+    RUNNING = 2
+    FINISH = 3
+
+
+class FranzGoVerifiableService(BackgroundThreadService):
+    """
+    FranzGoVerifiableService is si-verifier service.
+    To validate produced record user should run consumer and producer in one node.
+    Use ctx.cluster.alloc(ClusterSpec.simple_linux(1)) to allocate node and pass it to constructor
+    """
+    def __init__(self, context, redpanda, topic, msg_size, nodes):
+        self.custom_nodes = nodes is not None
+
+        if nodes:
+            num_nodes = 0
+        else:
+            num_nodes = 1
```

Comment:
can you write a description in the comment about the parameters. especially `nodes`. I'm not really sure what this initialization sequence based on `nodes` here is doing.